theseus/core/cost_weight.py

```diff
@@ -67,6 +73,9 @@ def __init__(
         self.scale.tensor = self.scale.tensor.view(-1, 1)
         self.register_aux_vars(["scale"])
 
+    def is_zero(self) -> torch.Tensor:
+        return (self.scale.tensor.abs() < EPS).view(-1)
```

Comment:
Do we allow negative `self.scale`?

Reply:
I agree this might speedup optimization. However, I'm afraid we might also lose some information for small `scale <EPS` but `scale != 0`. Not sure if this will raise some problems for differentiable optimization when we try to learn cost weights.

Reply:
I guess `_ZERO_EPS = 1.0e-15` is a reasonably small value to catch this? Is there a case where something in this range would indeed need to pass through as non zero?

Reply:
On the other hand, I think we can just use `tensor == 0`. Seems that torch's cutoff is about `7e-46` for floats and `1e-323` for doubles.

Reply:
Changed accordingly.